src/utils/utils.ts

```diff
@@ -223,32 +223,37 @@ function parseJSONRPCMessage<T extends JSONValue>(
  * Polykey errors are handled by their inbuilt `toJSON` method , so this only
  * serialises other errors
  */
-function replacer(key: string, value: any): any {
-  if (value instanceof AggregateError) {
-    // AggregateError has an `errors` property
-    return {
-      type: value.constructor.name,
-      data: {
-        errors: value.errors,
-        message: value.message,
-        stack: value.stack,
-      },
-    };
-  } else if (value instanceof Error) {
-    // If it's some other type of error then only serialise the message and
-    // stack (and the type of the error)
-    return {
-      type: value.name,
-      data: {
-        message: value.message,
-        stack: value.stack,
-      },
-    };
-  } else {
-    // If it's not an error then just leave as is
-    return value;
+
+let withinErrorKey = false;
+
+const replacer = (key: string, value: any) => {
+  if (key === 'error') {
+    withinErrorKey = true;
+  }
+  if (withinErrorKey && key !== 'code' && key !== 'message') {
+    if (value instanceof ErrorRPC) {
+      withinErrorKey = false;
+      return {
```

Comment:
The `withinErrorKey` here is essentially a global and will be shared across ALL usages of `replacer`. 

You want to be absolutely certain that's what you want if you're doing this. It's very likely that's undesired and you shouldn't be doing this.

Reply:
Good point, I'll handle the state in the function. Thanks!